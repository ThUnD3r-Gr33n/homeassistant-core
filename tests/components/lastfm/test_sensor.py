"""Tests for the lastfm sensor."""

from unittest.mock import AsyncMock

from pylast import WSError
from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant

from ..analytics_insights import setup_integration

from tests.common import MockConfigEntry


async def test_sensors(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_lastfm_network: AsyncMock,
    mock_lastfm_user: AsyncMock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test sensors."""
    await setup_integration(hass, mock_config_entry)

    entity_id = "sensor.lastfm_testaccount1"

    state = hass.states.get(entity_id)

    assert state == snapshot


async def test_first_time_user_state(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_lastfm_network: AsyncMock,
    mock_lastfm_user: AsyncMock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test first time user state."""
    mock_lastfm_user.get_playcount.return_value = 0
    mock_lastfm_user.get_recent_tracks.return_value = []
    mock_lastfm_user.get_top_tracks.return_value = []
    mock_lastfm_user.get_now_playing.return_value = None
    await setup_integration(hass, mock_config_entry)

    entity_id = "sensor.lastfm_testaccount1"

    state = hass.states.get(entity_id)

    assert state == snapshot


async def test_non_existent_user(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_lastfm_network: AsyncMock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test non existent user."""
    mock_lastfm_network.get_user.side_effect = WSError(
        "network", "status", "User not found"
    )
    await setup_integration(hass, mock_config_entry)

    entity_id = "sensor.lastfm_testaccount1"

    state = hass.states.get(entity_id)

    assert state == snapshot
