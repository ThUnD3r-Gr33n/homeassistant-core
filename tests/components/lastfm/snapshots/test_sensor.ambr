# serializer version: 1
# name: test_first_time_user_state
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Last.fm',
      'entity_picture': 'image',
      'friendly_name': 'LastFM testaccount1',
      'last_played': None,
      'play_count': 0,
      'top_played': None,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.lastfm_testaccount1',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'Not Scrobbling',
  })
# ---
# name: test_non_existent_user
  None
# ---
# name: test_sensors
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Last.fm',
      'entity_picture': 'image',
      'friendly_name': 'LastFM testaccount1',
      'last_played': None,
      'play_count': 1,
      'top_played': None,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.lastfm_testaccount1',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'artist - title',
  })
# ---
